### PR TITLE
Pietro Ballarin

### PR DESCRIPTION
**Pull request by pietroballarin**

### CODE REVIEW

File: routes/movies.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+router.get('/movies/edit/:id', (req, res, next) => {
+  Movie.findById(req.params.id)
+  .then(movie => {
+    console.log(movie)
+    res.render('movies/edit', { movieData: movie })
+  })
+  .catch(err => {
+    next(err);
+  })
+})
```
Comment (ahmedhany82) [inline routes/movies.js]: If you would like the user to be able to edit the cast as well, you can find all the celebrities from the database and pass it to res.render an object with two keys: `movieData` and `celebrityData`.

File: routes/movies.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+router.post('/movies/edit/:id', (req, res, next) => {
+  const movieId = req.params.id;
+  const {title, genre, plot} = req.body;
+  Movie.findByIdAndUpdate(movieId, {
+    title: title,
+    genre: genre,
+    plot: plot
+  })
```
Comment (ahmedhany82) [inline routes/movies.js]: Here you would consider updating the cast as well.

File: views/movies/edit.hbs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1>Edit a Movie</h1>
+{{#with movieData}}
+  <form action="/movies/edit/{{_id}}" method="POST">
+    <label for="title">Title </label>
+    <input type="text" name="title" id="title" value="{{title}}">
+    <label for="genre">Genre </label>
+    <input type="text" name="genre" id="genre" value="{{genre}}">
+    <label for="plot">Plot </label>
+    <input type="text" name="plot" id="plot" value="{{plot}}">
+    <button type="submit">Edit Movie</button>
```
Comment (ahmedhany82) [inline views/movies/edit.hbs]: Here you would also need to consider looping on the celebrities and figuring out how to show the ones in the movie cast as selected.

**Review by ahmedhany82 — COMMENTED** [review]
Hey Pietro,
Great work! Please find my comments below. Only related to the bonus. I would say give the bonus part a try.
Regards, Ahmed

File: routes/movies.js (hunk 1 above)
Comment (pietroballarin) [reply routes/movies.js]: Thanks for your feedback. I'll try to implement the iteration 12 as well :)

**Pull request closed without merging**